### DISCUSSION TITLE
BYD Battery NOT charging from AC with full (5KW) power

### DISCUSSION THREAD

**Discussion by AMichaelSuess** [binsentsu/home-assistant-solaredge-modbus#212] [General]
Dear all, also trying my luck here, as maybe somebody with a home battery can help. When I change the `Storage Remote Command Mode` to `Charge from PV and AC`, my battery charges with less than 1KW. It usually is able to charge with 5KW - and I have seen it doing that on solar power. I have the `Storage Remote Charge Limit` set to 5000W - and changing that also while the operation is running does nothing to increase the power. I have `Storage AC Charge Policy` set to `Always Allowed` - therefore also no `Storage AC Charge Limit` is set - or should be set according to the documentation. If I set one, it also does not make a difference. 

I have tried a thousand things already, logged into the installer WiFi-portal to see if anything was obviously misconfigured, tried to change some things back and forth - with no success. Also logged into the Battery-App from BYD - but it only allows updating the firmware - which has not changed anything.

While the charging is happening, also the values in the Solaredge-app look really strange - at the point when I took the picture, I had roughly 500W less house consumption - but why would the consumption of my house go up while charging the battery from the grid? When the timeout for the charge is up, the house consumption also goes down again accordingly to the 270W or so that was correct. 
![grafik](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/16519272/4f98f032-feda-47b8-9b3d-5d8e7b6a651e)

So what am I missing? How do I get my battery to charge with full 5KW power? If anybody in this round has any ideas or has seen this behaviour - I would greatly appreciate any pointers! Thanks!

**Comment by pdreijnders** [comment]
Hi Michael,
I only can help in supporting your issue. Because I have the same. Our setup is a SolarEdge invertor plus SolarEdge battery. 
So you can rule out the BYD battery as the source of the issue.
Regards,
Patrick

**Comment by AMichaelSuess** [comment]
Hmm, interesting. Just to further compare notes: I have an SE10K-RWS inverter. Can you share yours?

And: thanks for sharing - good to know I am not alone, at least this means my inverter is not broken - but at most misconfigured - or alternatively: feature-handicapped ;-).

**Reply by pdreijnders** [reply]
I have an SE5K-RWS inverter. It charges at around 2 kW at this moment.

**Comment by AMichaelSuess** [comment]
OK, found my problem. I had not set `Reactive Power mode to RRCR` - as is advised at the bottom of [this page](https://github.com/binsentsu/home-assistant-solaredge-modbus). I had only set `Set Advanced Power Control to Enable`. With this setting applied, my charge-rate went up to 5KW. Thanks for the help!

**Comment by marschallphilipp** [comment]
@AMichaelSuess  I have the same setup as you and I cannot load from AC since a few weeks. Before I could import from AC up to 5kW.
I´m not sure where I exactly set the Advanced Power Control and the Reactive Power mode. Can you help maybe?
I found the register values in this document on page 3
https://www.photovoltaikforum.com/core/attachment/88445-power-control-open-protocol-for-solaredge-inverters-pdf/
But how do I set this in the SetApp ? Under "Power Control" my "Power Reduction Interface (RRCR)" is disabled.

**Reply by AMichaelSuess** [reply]
Unfortunately, I don't remember what I did exactly in SetApp - and my inverter got replaced in the mean-time and the new one does not appear to need any interventions in SetApp anymore to properly charge with full power. What I did in a similar situation like yours - was carefully change one setting after the other, take screenshots of what I changed, tried to charge - see the result and if things did not improve, change the setting back. This way, I found the missing settings. Good luck!
